src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1252,6 +1252,10 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
         mark_broken();
         return res;
     }
+    catch (const Poco::TimeoutException &)
+    {
+        throw;
+    }
     catch (...)
     {
         mark_broken();
```

Comment:
This looks dangerous - why don't explicitly list the exceptions leading to broken parts?

Reply:
For example, what about the "memory limit exceeded" exception?

Reply:
> This looks dangerous

Yes, but it always worked this way. It's not that dangerous because it's easy to attach broken parts back

> why don't explicitly list the exceptions leading to broken parts?

I guess because we don't really know what exceptions lead to broken parts. Just like we don't know what exceptions are retriable. 

> what about the "memory limit exceeded" exception?

https://github.com/ClickHouse/ClickHouse/blob/eb698a7cdd168f12938e71a330b7208415c321b9/src/Storages/MergeTree/MergeTreeData.cpp#L1245-L1252
https://github.com/ClickHouse/ClickHouse/blob/eb698a7cdd168f12938e71a330b7208415c321b9/src/Storages/MergeTree/MergeTreeData.cpp#L1172-L1175